File: examples/python/schema-registry.py
from meshagent.api import SchemaRegistry, SchemaRegistration, MeshSchema, ElementType, ChildProperty, ValueProperty
import asyncio


sample = MeshSchema(
    root_tag_name="sample",
    elements=[
        ElementType(
            description="test",
            tag_name="sample",
            properties=[
                ChildProperty(name="children", description="desc", child_tag_names=[
                    "child"
                ]),
        ]),
        ElementType(tag_name="child", description="child", properties=[
            ValueProperty(name="prop", description="desc", type="number")
        ]),
    ]
)

async def main():    
    server = SchemaRegistry(schemas=[
        SchemaRegistration(name="sample", schema=sample)
    ])
    await server.run()

asyncio.run(main())

File: examples/typescript/schema-registry.ts
import {
  SchemaRegistry,
  SchemaRegistration,
  MeshSchema,
  ElementType,
  ChildProperty,
  ValueProperty,
} from '@meshagent/meshagent';

// Define the schema
const sample = new MeshSchema({
  rootTagName: 'sample',
  elements: [
    new ElementType({
      tagName: 'sample',
      description: 'test',
      properties: [
        new ChildProperty({
          name: 'children',
          description: 'desc',
          childTagNames: ['child'],
        }),
      ],
    }),
    new ElementType({
      tagName: 'child',
      description: 'child',
      properties: [
        new ValueProperty({
          name: 'prop',
          description: 'desc',
          type: 'number',
        }),
      ],
    }),
  ],
});

// Main async function to run the server
async function main(): Promise<void> {
  const server = new SchemaRegistry({
    schemas: [
      new SchemaRegistration({
        name: 'sample',
        schema: sample,
      }),
    ],
  });

  // Start the server
  await server.run();
}

// Entry point
try {
  main();
} catch (error) {
  console.error('Error:', error);
}
